pandera/dtypes.py

```diff
@@ -1,458 +1,484 @@
-# pylint: disable=no-member,too-many-public-methods
-"""Schema datatypes."""
-
-from enum import Enum
-from typing import Optional, Union
-
-import numpy as np
-import pandas as pd
-from packaging import version
-
-PandasExtensionType = pd.core.dtypes.base.ExtensionDtype
-
-LEGACY_PANDAS = version.parse(pd.__version__).major < 1  # type: ignore
-NUMPY_NONNULLABLE_INT_DTYPES = [
-    "int",
-    "int_",
-    "int8",
-    "int16",
-    "int32",
-    "int64",
-    "uint8",
-    "uint16",
-    "uint32",
-    "uint64",
-]
-
-NUMPY_TYPES = frozenset(
-    [item for sublist in np.sctypes.values() for item in sublist]  # type: ignore
-).union(
-    frozenset([np.complex_, np.int_, np.uint, np.float_, np.str_, np.bool_])
+"""Pandera data types."""
+# pylint:disable=too-many-ancestors
+import dataclasses
+import inspect
+from abc import ABC
+from typing import (
+    Any,
+    Callable,
+    Iterable,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
 )
 
-# for int and float dtype, delegate string representation to the
-# default based on OS. In Windows, pandas defaults to int64 while numpy
-# defaults to int32.
-_DEFAULT_PANDAS_INT_TYPE = str(pd.Series([1]).dtype)
-_DEFAULT_PANDAS_FLOAT_TYPE = str(pd.Series([1.0]).dtype)
-_DEFAULT_PANDAS_COMPLEX_TYPE = str(pd.Series([complex(1)]).dtype)
-_DEFAULT_NUMPY_INT_TYPE = str(np.dtype(int))
-_DEFAULT_NUMPY_FLOAT_TYPE = str(np.dtype(float))
 
+class DataType(ABC):
```

Comment:
it looks like `dtypes.py` and `dtypes_.py` implement a bunch of the same classes/functions. Is this intentional?

Reply:
whoops, `dtypes_.py` is deleted on my machine but I did not add the removal to the commit. 
Fixed now.